libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/oauth/token/DefaultOAuthJwtAccessToken.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.auth.oauth.token;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.RequiredTypeException;
+
+/**
+ * Default implementation of OAuthJwtAccessToken
+ */
+public class DefaultOAuthJwtAccessToken implements OAuthJwtAccessToken {
+
+    // protected JwsHeader<?> header;
+    protected Claims body;
+    protected String signature;
+
+    /**
+     * Create DefaultOAuthJwtAccessToken access token object
+     * @param  jws JWS claims
+     */
+    public DefaultOAuthJwtAccessToken(Jws<Claims> jws) {
+        // this.header = jws.getHeader();
+        this.body = jws.getBody();
+        this.signature = jws.getSignature();
+    }
+
+    @Override
+    public String getSubject() {
+        String subject = this.body.getSubject();
+        if (subject == null) {
+            return null;
+        }
+        return subject.toLowerCase(); // normalize as Athenz principal
+    }
+
+    @Override
+    public String getIssuer() {
+        return this.body.getIssuer();
+    }
+
+    @Override
+    public String getAudience() {
+        // aud can be string or an array of strings.
+        return this.body.getAudience();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<String> getAudiences() {
+        // https://tools.ietf.org/html/rfc7519#page-9
+        List<String> audiences;
+        try {
+            // returns null if not found
+            audiences = this.body.get(Claims.AUDIENCE, ArrayList.class);
+        } catch (RequiredTypeException e) {
```

Comment:
we should probably log these errors otherwise it's very difficult to debug any issues.

Reply:
added in 24f02031